changelog/changes/2025-04-03-sanitize-hostname.md

```diff
@@ -0,0 +1,2 @@
+* Always truncate hostnames on the first occurrence of `.`
+* Ensure hostnames never exceeds 63 characters, regardless of the metadata provider
```

Comment:
Please replace `*` with `-` for consitency, and would be great if you can include a link to the PRs eg ([coreos-cloudinit#31](http://github.com/flatcar/coreos-cloudinit/pull/31))